src/types/array.js

```diff
@@ -41,6 +41,16 @@ export default parameterized(T => class ArrayType {
     return [value, ...valueOf(this)];
   }
 
+  sort(compareFn) {
+    let init = valueOf(this);
+    let result = [...init].sort(compareFn);
+    let equalityCheck = comparatorList =>
+      init.every((member, index) =>
+        comparatorList[index] === member);
+
+    return equalityCheck(result) ? this : result;
```

Comment:
Now that it's this concise, it should fit within 80 characters, so let's just inline it if we can..